code/modules/mob/living/carbon/human/human_attackhand.dm

```diff
@@ -346,22 +352,30 @@
 			msg_admin_attack("[key_name(M)] disarmed [src.name] ([src.ckey]) (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[M.x];Y=[M.y];Z=[M.z]'>JMP</a>)",ckey=key_name(M),ckey_target=key_name(src))
 			M.do_attack_animation(src)
 
+			M.stamina = M.stamina - disarm_cost //attempting to knock something out of someone's hands, or pushing them over, is exhausting!
+			M.stamina = Clamp(M.stamina, 0, M.max_stamina)
+
 			if(w_uniform)
 				w_uniform.add_fingerprint(M)
 			var/obj/item/organ/external/affecting = get_organ(ran_zone(M.zone_sel.selecting))
 
-			var/list/holding = list(get_active_hand() = 40, get_inactive_hand = 20)
+			var/list/holding = list(get_active_hand() = 40, get_inactive_hand() = 20)
 
-			//See if they have any guns that might go off
-			for(var/obj/item/weapon/gun/W in holding)
+			//See if they have any weapons to retaliate with
+			for(var/obj/item/weapon/W in holding)

```

Comment:
Please. PLEASE make this depend on intent. I don't want to decapitate someone for lightly shoving me.